packages/react-composites/tests/browser/chat/fixture.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PARTICIPANT_NAMES } from '../config';
+import { IdentityType, createChatThreadAndUsers, loadPage } from '../utils';
+import { startServer, stopServer } from './app/server';
+import { chromium, Browser, Page, test as base } from '@playwright/test';
+import dotenv from 'dotenv';
+import path from 'path';
+
+dotenv.config({ path: path.join(__dirname, '..', '.env') });
+
+export interface ChatWorkerFixtures {
+  serverUrl: string;
+  testBrowser: Browser;
+  users: IdentityType[];
+  pages: Array<Page>;
+}
+
+/**
+ * A worker-scoped test fixture for ChatComposite browser tests.
+ */
+export const test = base.extend<unknown, ChatWorkerFixtures>({
+  /**
+   * Run a server to serve the ChatComposite example app.
+   *
+   * @returns string URL for the server.
+   */
+  serverUrl: [
+    // playwright forces us to use a destructuring pattern for first argument.
+    /* eslint-disable-next-line no-empty-pattern */
+    async ({}, use) => {
+      await startServer();
+      try {
+        await use(SERVER_URL);
+      } finally {
+        await stopServer();
+      }
+    },
+    { scope: 'worker' }
+  ],
+  /**
+   * Starts a test browser to load the ChatComposite example app.
+   *
+   * @returns Browser object.
+   */
+  testBrowser: [
+    // playwright forces us to use a destructuring pattern for first argument.
+    /* eslint-disable-next-line no-empty-pattern */
+    async ({}, use) => {
+      const browser = await chromium.launch({
+        args: ['--start-maximized', '--disable-features=site-per-process'],
+        headless: true
+      });
+      try {
+        await use(browser);
+      } finally {
+        await browser.close();
+      }
+    },
+    { scope: 'worker' }
+  ],
+  /**
+   * Creates a Chat thread and adds users to the thread.
+   *
+   * @returns the created users' identities.
+   */
+  users: [
+    // playwright forces us to use a destructuring pattern for first argument.
+    /* eslint-disable-next-line no-empty-pattern */
+    async ({}, use) => {
+      const users = await createChatThreadAndUsers(PARTICIPANTS);
+      await use(users);
+    },
+    { scope: 'worker' }
+  ],
+  /**
+   * Loads the ChatComposite example app for each participant in a browser page.
+   *
+   * @returns Array of Page's loaded, one for each participant.
+   */
+  pages: [
+    async ({ serverUrl, testBrowser, users }, use) => {
+      const pages = await Promise.all(users.map(async (user) => loadPage(testBrowser, serverUrl, user)));
+      await use(pages);
+    },
+    { scope: 'worker' }
+  ]
+});
+
+const SERVER_URL = 'http://localhost:3000';
```

Comment:
nit: just a personal taste to have those const declared at the beginning not the end

Reply:
Conflicts with the intent to declare public stuff up top.

A reader only intending to use this package doesn't need to know about `SERVER_URL`, so I don't want to show it to them first.

Reply:
Moved it up to make progress. I don't care too much about this. But the intent was to keep clutter off a reader who only cares about public API of the package ( i.e., the important reader ;) )